packages/flame/lib/src/components/mixins/collidable.dart

```diff
@@ -28,6 +30,21 @@ mixin Collidable on Hitbox {
     }
     super.onRemove();
   }
+
+  @override
+  @mustCallSuper
+  void prepare(Component parent) {
+    super.prepare(parent);
+
+    if (isPrepared) {
```

Comment:
so if I am understanding this, if isPrepared is false it means this is being add as a child from a component that has not yet been added to any game.

and then later when that component is added to a game, it and its child (this) will be prepared again.

is that right? if so I think having a test for this case is important as it's non trivial but seems easy to test and make sure it's working

Reply:
Yeah, that is what I understood from my investigation too